Offer display permissions

wallyworld opened the pull request:
## Description of change

This PR builds on https://github.com/juju/juju/pull/7167
Please only review the second commit https://github.com/wallyworld/juju/commit/6f26818d07323b4ee5227d1b96f8faf94071803b
(Show offer permission in CLI output)

We now show the application offer permission a user has when running juju find-endpoints of show-endpoints.

## QA steps

`$juju find-endpoints 
URL                     Access   Interfaces
admin/controller.mysql  read     mysql:db
admin/controller.foo    consume  mysql:db

$juju show-endpoints admin/controller.mysql
URL                     Access   Description                                    Endpoint  Interface  Role
admin/controller.mysql  consume  MySQL is a fast, stable and true multi-user,   db        mysql      provider
                                 multi-threaded SQL database server. SQL                             
                                 (Structured Query Language) is the most                             
                                 popular database query language in the world.                       
                                 The ma...`

axw commented:
LGTM, I think it would be best to have a unit test that covers the NotFound offer access case.

wallyworld commented:
The mock was wrongly implemented - I fixed it to reflect the correct behavour and unit tests which exercise the functionality pass with the code change Also added a new bit of testing to the state package.

wallyworld commented:
$$merge$$
